src/lib/variants.ts

```diff
@@ -251,6 +254,42 @@ export const hasNoTicker: Filter = {
     },
 };
 
+// Edge case filter to exclude any test where the combination of targeting data and variants logic
+// would allow the test to pass, but the value used to replace the article count
+// in the Epic copy is "0" (zero) due to all article views having been
+// within the`periodInWeeks` defined in the test (used for targeting)
+// but before (i.e. older) than the hardcoded `52` weeks (used for template rendering).
+// Prevents an Epic from ever rendering with text "...you've read 0 articles...".
+// This is needed because we use different timeframes when finding a
+// test/variant and when replacing the Epic placeholder, so we should
+// have a specific check around the logic/values used for template rendering.
+// This is intentional and in parity with Frontend.
+export const hasNoZeroArticleCount = (now: Date = new Date()): Filter => ({
```

Comment:
I would go for a shorter comment here. Something like:

`Prevent cases like "...you've read 0 articles...".`

That captures the key motivation. If someone does want to understand further, they can just look at the code (which itself is well commented).

If you do want a lengthier form, I think you could still keep this shorter and clearer. E.g.

`Prevent cases like "...you've read 0 articles...". This could happen when the article history required by the test is different than the date range used by the template itself (which is always 52 weeks at the time of writing).`

Reply:
Cool - updated